packages/webapp-libs/webapp-tenants/src/components/tenantSwitch/tenantSwitch.component.tsx

```diff
@@ -31,17 +31,47 @@ export const TenantSwitch = () => {
 
   const tenantsGrouped = groupBy(prop<string>('type'), tenants);
   const personalTenant = head(tenantsGrouped[TenantType.PERSONAL]);
-  const organizationTenants = tenantsGrouped[TenantType.ORGANIZATION] ?? [];
+  const organizationTenants = groupBy(
+    (tenant) => (tenant?.membership?.invitationAccepted ? 'organizations' : 'invitations'),
+    tenantsGrouped[TenantType.ORGANIZATION] ?? []
+  );
 
-  const handleTenantChange = (tenant?: TenantListItemFragmentFragment | null) => () => {
+  const handleTenantChange = (tenant?: CommonQueryTenantItemFragmentFragment | null) => () => {
     if (!tenant) return;
     navigate(generateTenantPath(RoutesConfig.home, { tenantId: tenant.id }));
   };
 
+  const handleInvitationClick = (tenant?: CommonQueryTenantItemFragmentFragment | null) => () => {
+    const token = tenant?.membership?.invitationToken;
+    if (!token) return;
+    navigate(generateLocalePath(RoutesConfig.tenantInvitation, { token }));
+  };
+
   const handleNewTenantClick = () => {
     navigate(generateLocalePath(RoutesConfig.addTenant));
   };
 
+  const handleTenantSettingsClick = () => {
+    navigate(generateTenantPath(TenantRoutesConfig.tenant.settings.members));
+  };
+
+  const handleLastInvitationClick = () => {
+    handleInvitationClick(organizationTenants?.invitations?.[0])();
+  };
+
+  const renderPendingInvitationBadge = () => {
+    return (
+      <Button color="primary" size="sm" onClick={handleLastInvitationClick}>
+        <UserPlus className="mr-2" size="16" />
+        <FormattedMessage
+          defaultMessage="{invitationsCount, plural, =1 {# invitation} other {# invitations}}"
+          id="TenantSwitch / Invitations Badge"
+          values={{ invitationsCount: organizationTenants?.invitations?.length }}
```

Comment:
maybe we could use some kind of fallback values in case of undefined? `values={{ invitationsCount: organizationTenants?.invitations?.length ?? 0 }}`.